Whiskey/Tasks/Invoke-WhiskeyExec.ps1

```diff
@@ -120,19 +116,87 @@ function Invoke-WhiskeyExec
+    $successExitRangeLessThan = $null
+
     if ( $TaskParameter['SuccessExitCode'] )
     {
-        $successExitCode = $TaskParameter['SuccessExitCode']
+        $TaskParameter['SuccessExitCode'] | ForEach-Object {
+            
+            if( $_ -match '^\d+$' )
+            {
+                $successExitCode += $_
+            }
+            
+            elseif( $_ -match '^\=\s*\d+$' )
+            {
+                $successExitCode += [int]($_ -split '(\d+$)')[1]
+            }
+            
+            elseif( $_ -match '^\d+\.\.\d+$' )
+            {
+                $successExitCode += Invoke-Expression -Command $_
+            }
+            
+            elseif ($_ -match '^>\s*\d+$')
+            {
+                $gtTemp = [int]($_ -split '(\d+$)')[1]
+                if ( !$successExitRangeGreaterThan -or $gtTemp -lt $successExitRangeGreaterThan )
+                {
+                    $successExitRangeGreaterThan = $gtTemp
+                }
+            }
+
+            elseif ($_ -match '^>\=\s*\d+$')
+            {
+                $gteTemp = [int]($_ -split '(\d+$)')[1] - 1
+                if ( !$successExitRangeGreaterThan -or $gteTemp -lt $successExitRangeGreaterThan )
+                {
+                    $successExitRangeGreaterThan = $gteTemp
+                }
+            }
+
+            elseif ($_ -match '^<\s*\d+$')
+            {
+                $ltTemp = [int]($_ -split '(\d+$)')[1]
+                if ( !$successExitRangeLessThan -or $ltTemp -lt $successExitRangeLessThan )
+                {
+                    $successExitRangeLessThan = $ltTemp
+                }
+            }
+
+            elseif ($_ -match '^<\=\s*\d+$')
+            {
+                $lteTemp = [int]($_ -split '(\d+$)')[1] + 1
+                if ( !$successExitRangeLessThan -or $lteTemp -lt $successExitRangeLessThan )
+                {
+                    $successExitRangeLessThan = $lteTemp
+                }
+            }
+        }
     }
-
-
+    else
+    {
+        $successExitCode = 0
+    }
+    
     $process = Start-Process -FilePath $path @argumentListParam -WorkingDirectory $workingDirectory -NoNewWindow -Wait -PassThru
-
     $exitCode = $process.ExitCode
-    if ( $exitCode -notin $successExitCode )
+    
+    if ( !$successExitRangeGreaterThan )
+    {
+        $successExitRangeGreaterThan = $exitCode + 1
+    }
+    
+    if ( !$successExitRangeLessThan )
+    {
+        $successExitRangeLessThan = $exitCode - 1
+    }
+
+    if ( !($exitCode -in $successExitCode) -and !($exitCode -gt $successExitRangeGreaterThan) -and !($exitCode -lt $successExitRangeLessThan) )
     {
         Stop-WhiskeyTask -TaskContext $TaskContext -Message ('''{0}'' returned with an exit code of ''{1}'', which is not one of the expected ''SuccessExitCode'' of ''{2}''. View the build output to see why the executable''s process failed.' -F $TaskParameter['Path'],$exitCode,$successExitCode -join ',')

```

Comment:
Change this to not mention the success exit codes.

Reply:
Updated